_plugins/convert_md.rb

```diff
@@ -1,9 +1,9 @@
 # Convert the links that point to a 'README.md' to 'index.html'
 Jekyll::Hooks.register :pages, :post_render do |doc|
-  doc.output = doc.output.gsub(/(href="\.?\.\/[^.]*)README\.md?(#(.*))?"/, '\1index.html\2"')
+  doc.output = doc.output.gsub(/(href="\.?\.?\/?\.?\.?\/?\.?\.\/[^.]*)README\.md?(#(.*))?"/, '\1index.html\2"')
```

Comment:
```suggestion
  doc.output = doc.output.gsub(/(href="(\.\/|(\.\.\/)+)[^.]*)README\.md?(#(.*))?"/, '\1index.html\2"')
```

To match `./` or `../` or `../../../../../../` etc

Reply:
I've tested this suggestion and it seems some links got broken, will revisit this regex in the future to see if there's a better way to handle these nested levels